### PR TITLE
Feat: feat(kpm): support command 'kpm add' from github. #4.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #4 

#### 2. What is the scope of this PR (e.g. component or file name):

kpm add

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

support command 'kpm add' from github.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/opt/opt.go [hunk 1]
```diff
@@ -7,3 +7,19 @@ type InitOptions struct {
 	Name     string
 	InitPath string
 }
+
+type AddOptions struct {
+	LocalPath    string
+	RegistryOpts RegistryOption
+}
+
+type RegistryOption struct {
```
Comment (Peefy) [inline pkg/opt/opt.go:16]: s/RegistryOption/RegistryOptions

File: pkg/opt/opt.go [hunk 2]
```diff
@@ -7,3 +7,19 @@ type InitOptions struct {
...
+	Git *GitOption
+}
+
+type GitOption struct {
```
Comment (Peefy) [inline pkg/opt/opt.go:20]: s/GitOption/GitOptions

Comment (zong-zhe) [reply pkg/opt/opt.go]: fixed.

File: pkg/opt/opt.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/opt/opt.go]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Comment by zong-zhe** [comment]
Commit 99ae93e759037bd1ff9cdc44cb640431ed8fd44b will save the downloaded different versions of the kcl package in KPM_HOME as described in issue https://github.com/KusionStack/kpm/issues/21.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**